include/class.thread.php

```diff
@@ -1085,12 +1085,12 @@ protected function normalizeFileInfo($files, $add_error=true) {
         foreach ($files as $id => $info) {
             $F = array('inline' => is_array($info) && @$info['inline']);
 
-            if (is_numeric($id) && $id != 0)
+            if (is_array($info) && isset($info['id']))
+                $fileId = $info['id'];
+            elseif (is_numeric($id) && $id != 0)
```

Comment:
I wonder if we still need this case - we should track down possibles just to make sure we don't end up with wrong attachments again.

Reply:
One scenario you didn't cover is email attachments processed via mail parse (piped emails). This would result in $info being array of file data to be created! Since $info['id'] doesn't exist yet - we end up with wrong references.